### PR TITLE
Feature/tdp 3331 version a preparation and list versions

### PR DESCRIPTION
**Pull request by amaalej**
**Link to the JIRA issue**
e.g. https://jira.talendforge.org/browse/TDP-3331

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)

**(Optional) Other information**:

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/59/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/59/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/64/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/64/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/68/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/68/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/69/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/69/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/85/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/85/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/88/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/88/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/94/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/94/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/97/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/97/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/99/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/99/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1/)

File: dataprep-webapp/src/app/components/playground/_playground.scss [hunk 1]
```diff
@@ -175,15 +175,37 @@ $playground-height: '100vh - ' + ($navigation-all-height);
...
-					@include calc('height', $playground-suggestions-stats-details);
+				.stat-detail-tab-items {
 					width: 330px;
+				}
+			}
 
-					.stat-detail-tab-items {
-						width: 330px;
+			#help-suggestions {
+				overflow-y: hidden;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/playground/_playground.scss]: Remove this hack and fix it properly please ....

File: dataprep-webapp/src/app/components/playground/_playground.scss [hunk 2]
```diff
@@ -175,15 +175,37 @@ $playground-height: '100vh - ' + ($navigation-all-height);
...
+				.loading-container {
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/playground/_playground.scss]: Remove this level
(adapt the html removing `<div class="loading-container">`)

File: dataprep-webapp/src/app/components/playground/_playground.scss [hunk 3]
```diff
@@ -175,15 +175,37 @@ $playground-height: '100vh - ' + ($navigation-all-height);
...
+					position: relative;
+
+					.loading-msg {
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/playground/_playground.scss]:
```
.loading-msg {
	position: absolute;
	text-align: center;
	top: 100px;
	left: 0;
	right: 0;

	background: transparent;
	color: $pale-blue;
	opacity: 0.8;
}
```

File: dataprep-webapp/src/app/components/playground/_playground.scss [hunk 4]
```diff
@@ -175,15 +175,37 @@ $playground-height: '100vh - ' + ($navigation-all-height);
...
-				.suggestions-stats-content {
-					@include calc('height', $playground-suggestions-stats-details);
+				.stat-detail-tab-items {
 					width: 330px;
+				}
+			}
 
-					.stat-detail-tab-items {
-						width: 330px;
+			#help-suggestions {
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/playground/_playground.scss]: This should not be here, this is action-suggestions

File: dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js [hunk 5]
```diff
@@ -12,7 +12,7 @@
   ============================================================================*/
 const SuggestionsStats = {
 	template: `
-    <div class="suggestions-stats-details">
+    <div class="actions-header-details">
 		<div class="suggestions-title" ng-switch="suggestionsStatsCtrl.state.playground.grid.selectedColumns.length > 1">
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js]: Why this extra div ? I think you can remove it, and adapt the style if needed

File: dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js [hunk 6]
```diff
@@ -12,7 +12,7 @@
   ============================================================================*/
 const SuggestionsStats = {
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js]: Rename please

File: dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js [hunk 7]
```diff
@@ -12,7 +12,7 @@
...
 	template: `
-    <div class="suggestions-stats-details">
+    <div class="actions-header-details">
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js]: Why actions-header`-details`?

File: dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js [hunk 8]
```diff
@@ -26,7 +26,6 @@ const SuggestionsStats = {
 				  translate-values="{nb: suggestionsStatsCtrl.state.playground.grid.selectedColumns.length}">
 			</span>
 		</div>
-		<ng-transclude></ng-transclude>
     </div>`,
 	transclude: true,
 	controllerAs: 'suggestionsStatsCtrl',
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js]: Rename please

File: dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js [hunk 9]
```diff
@@ -26,7 +26,6 @@ const SuggestionsStats = {
 				  translate-values="{nb: suggestionsStatsCtrl.state.playground.grid.selectedColumns.length}">
 			</span>
 		</div>
-		<ng-transclude></ng-transclude>
     </div>`,
 	transclude: true,
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js]: Remove please

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/18/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/18/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/20/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/20/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/21/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/21/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/22/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/22/)

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 10]
```diff
@@ -832,7 +833,7 @@ export default function PlaygroundService($state, $rootScope, $q, $translate, $t
 	 * @methodOf data-prep.services.playground.service:PlaygroundService
 	 * @description open a preparation
 	 */
-	function initPreparation() {
+	function initPreparation(prepid) {
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: done

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 11]
```diff
@@ -862,10 +863,10 @@ export default function PlaygroundService($state, $rootScope, $q, $translate, $t
 	 * @methodOf data-prep.services.playground.service:PlaygroundService
 	 * @description open a dataset
 	 */
-	function initDataset() {
+	function initDataset(datasetid) {
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: done

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/24/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/24/)

File: dataprep-webapp/src/app/services/playground/playground-service.js (hunk 10 above)
Comment (jsomsanith) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: The unit tests are still not cleaned : $stateParams is not needed anymore

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js [hunk 12]
```diff
@@ -60,129 +60,227 @@ describe('CollapsiblePanel container', () => {
...
+				header: [simpleItem, badgeItem],
+				content: [],
+			};
+
+			// when
+			createElement();
+
+			// then
+			expect(element.find('.panel-heading > div > div').eq(0).text().trim()).toBe(simpleItem.label);
+			expect(element.find('.panel-heading > div > div').eq(1).text().trim()).toBe(badgeItem.label);
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: Still KO

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js [hunk 13]
```diff
@@ -60,129 +60,227 @@ describe('CollapsiblePanel container', () => {
...
+					}, {
+						label: 'by Henry-Mayeul de Benque',
+						bsStyle: 'default',
+						tooltipPlacement: 'top',
+						className: 'text-right',
+					},
+				],
+				description: `Lorem ipsum`
+			},
+			onSelect: 'version:select',
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: Still ko

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js [hunk 14]
```diff
@@ -60,129 +60,227 @@ describe('CollapsiblePanel container', () => {
...
+			onToggle: 'version:toggle',
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: Same here

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/34/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/34/)

File: dataprep-webapp/src/app/components/playground/_playground.scss (hunk 4 above)
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/playground/_playground.scss]: Still KO

File: dataprep-webapp/src/app/components/playground/_playground.scss (hunk 1 above)
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/playground/_playground.scss]: Still KO

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js (hunk 12 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: done

File: dataprep-webapp/src/app/components/playground/_playground.scss (hunk 2 above)
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/playground/_playground.scss]: Still KO

File: dataprep-webapp/src/app/components/playground/_playground.scss (hunk 3 above)
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/playground/_playground.scss]: Still ko

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js (hunk 13 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: done

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js (hunk 14 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: done

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js [hunk 15]
```diff
@@ -60,129 +60,227 @@ describe('CollapsiblePanel container', () => {
...
+			theme: 'descriptive-panel',
 		};
 
-		// when
-		createElement();
+		it('should render a selectable header', () => {
```
Comment (amaalej) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: done

File: dataprep-webapp/src/app/components/playground/header/_playground-header.scss [hunk 16]
```diff
@@ -68,23 +68,29 @@ $playground-preview-icon-padding: 10px !default;
 		flex-grow: 1;
 	}
 
-	ul > li {
-		display: flex;
-		align-items: center;
+	ul {
+		> * {
```
Comment (amaalej) [inline dataprep-webapp/src/app/components/playground/header/_playground-header.scss]: done

File: dataprep-webapp/src/app/components/playground/header/playground-header.html [hunk 17]
```diff
@@ -59,7 +59,7 @@
 </nav>
 <nav class="playground-sub-header navbar">
     <ul class="playground-sub-header-left nav navbar-nav">
-        <li ng-switch="!!$ctrl.isReadonly">
+        <li ng-switch="!!$ctrl.isReadonly" insertion-sub-header-left>
```
Comment (amaalej) [inline dataprep-webapp/src/app/components/playground/header/playground-header.html]: done

File: dataprep-webapp/src/app/components/playground/header/playground-header.html [hunk 18]
```diff
@@ -81,7 +81,7 @@
             <span translate-once="READ_ONLY"></span>
         </li>
     </ul>
-    <ul class="playground-sub-header-right nav navbar-nav">
+    <ul class="playground-sub-header-right nav navbar-nav" insertion-manage-versions>
```
Comment (amaalej) [inline dataprep-webapp/src/app/components/playground/header/playground-header.html]: done

File: dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js (hunk 8 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js]: done

File: dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js (hunk 9 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js]: done

File: dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js (hunk 5 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js]: done

File: dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js (hunk 7 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js]: done

File: dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js (hunk 6 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/suggestions-stats/actions-header/actions-header-component.js]: done

File: dataprep-webapp/src/app/components/playground/_playground.scss (hunk 3 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/playground/_playground.scss]: done

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/39/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/39/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/40/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/40/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/41/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/41/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/44/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/44/)

File: dataprep-webapp/src/app/services/playground/playground-service.js (hunk 10 above)
Comment (charlesnguyen) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: instead $stateParams is needed for shouldReloadPreparation function

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/50/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/50/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/51/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/51/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/52/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/52/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/54/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/54/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/57/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/57/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/61/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/61/)

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-controller.js [hunk 19]
```diff
@@ -22,23 +22,32 @@ export default class CollapsiblePanelCtrl {
...
-	$onChanges() {
+	$onChanges(changes) {
+		if (!changes.item) {
+			return;
+		}
+		this.adaptItem();
+	}
+
+	adaptItem() {
+		this.adaptedItem = { ...this.item };
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-controller.js]: Why did you not use `this.adaptedItem = this.item;`

File: dataprep-webapp/src/app/services/playground/playground-service.spec.js [hunk 20]
```diff
@@ -2155,18 +2162,16 @@ describe('Playground Service', () => {
 			spyOn(PlaygroundService, 'loadDataset').and.returnValue($q.when());
 
 			// when
-			PlaygroundService.initDataset();
+			PlaygroundService.initDataset('dataset1');
 			$rootScope.$apply();
 
 			// then
-			expect(PlaygroundService.loadDataset).toHaveBeenCalled();
+			expect(PlaygroundService.loadDataset).toHaveBeenCalledWith('dataset1');
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.spec.js]: Use var instead of litteral

File: dataprep-webapp/src/app/services/playground/playground-service.spec.js [hunk 21]
```diff
@@ -1957,6 +1956,15 @@ describe('Playground Service', () => {
 			);
 		}));
 
+		it('should call get dataset details', inject(($q, $rootScope, PlaygroundService, PreparationService) => {
+			// when
+			PlaygroundService.initPreparation('prep1');
+			$rootScope.$apply();
+
+			// then
+			expect(PreparationService.getDetails).toHaveBeenCalledWith('prep1');
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.spec.js]: Use var instead of litteral

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 22]
```diff
@@ -73,6 +73,7 @@ export default function PlaygroundService($state, $rootScope, $q, $translate, $t
 		loadStep,           // load preparation step
 		updateStatistics,   // load column statistics and trigger statistics update
 		close,
+		errorGoBack,
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: Why did you add it?

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js [hunk 23]
```diff
@@ -60,129 +60,311 @@ describe('CollapsiblePanel container', () => {
...
+			it('should render body header', () => {
+				// given
+				scope.version = version;
+
+				// when
+				createElement();
+
+				// then
+				expect(element.find('.panel-body > div').eq(0).find('span').eq(0).text().trim()).toBe(version.content.head[0].label);
+				expect(element.find('.panel-body > div').eq(0).find('span').eq(1).text().trim()).toBe(version.content.head[1].label);
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: Use var instead parsing N times DOM for `element.find('.panel-body > div').eq(0).find('span')`

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js [hunk 24]
```diff
@@ -60,129 +60,311 @@ describe('CollapsiblePanel container', () => {
...
+				};
+
+				// when
+				createElement();
+
+				// then
+				expect(element.find('.panel-heading > div').length).toBe(1);
+
+				expect(element.find('.panel-heading > div').eq(0).find('span').eq(0).text().trim()).toBe(simpleItem.label);
+				expect(element.find('.panel-heading > div').eq(0).find('span').eq(1).text().trim()).toBe(badgeItem.label);
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: Use var instead parsing N times DOM for `element.find('.panel-heading > div').eq(0).find('span')`

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js [hunk 25]
```diff
@@ -60,129 +60,311 @@ describe('CollapsiblePanel container', () => {
...
+					content: [],
+				};
+
+				// when
+				createElement();
+
+				// then
+				expect(element.find('.panel-heading > div > div').eq(0).text().trim()).toBe(simpleItem.label);
+				expect(element.find('.panel-heading > div > div').eq(1).text().trim()).toBe(badgeItem.label);
+				expect(element.find('.panel-heading > div > div').eq(1).find('span').hasClass('label')).toBe(true);
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: Use var instead parsing N times DOM for `element.find('.panel-heading > div > div')`

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-controller.js (hunk 19 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-controller.js]: Because I need to change the object 
`this.adaptedItem.onToggle = event => headerOnToggle(event, this.adaptedItem);` 
`this.adaptedItem.onSelect = event => headerOnSelect(event, this.adaptedItem);`
and I shouldn't change the initial object

File: dataprep-webapp/src/app/services/playground/playground-service.spec.js (hunk 21 above)
Comment (amaalej) [inline dataprep-webapp/src/app/services/playground/playground-service.spec.js]: done

File: dataprep-webapp/src/app/services/playground/playground-service.spec.js (hunk 20 above)
Comment (amaalej) [inline dataprep-webapp/src/app/services/playground/playground-service.spec.js]: ok

File: dataprep-webapp/src/app/services/playground/playground-service.js (hunk 22 above)
Comment (amaalej) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: it's used in ee

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js (hunk 25 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: ok

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js (hunk 24 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: ok

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js (hunk 23 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-container.spec.js]: ok

File: dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-controller.js (hunk 19 above)
Comment (amaalej) [inline dataprep-webapp/src/app/components/widgets-containers/collapsible-panel/collapsible-panel-controller.js]: besides, it was a solution to get around a synchronisation problem between the react component and angular

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/87/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/87/)
